### ISSUE TITLE
[Bug] ETP dropdown has bad TalkBack semantics

### ISSUE THREAD

**Issue by eeejay** [mozilla-mobile/fenix#8589]
* The switch and label should be grouped together so you hear "Enhanced Tracking Protection, Switch ON". Now those are two separate items, so you get an unlabeled switch (ie. "Switch, ON").
* The "Blocked" heading should be marked as a heading.
* Activating the blocked items goes to another screen but TalkBack does not set the cursor inside it.
* The back button in the "blocked items" info sheet is unlabeled.
* The "Blocked" heading in the info sheet also needs a heading role.
* Each of the blocked trackers should be distinct and not be read all at once.

**Comment by codrut-topliceanu** [comment]
@eeejay 

> Each of the blocked trackers should be distinct and not be read all at once.

Do you mean it should be as a list view where the user has to navigate through the list's elements? I'm asking because there are sites were we have over a dozen or so blocked trackers.

**Comment by asavill** [comment]
Is this related to https://github.com/mozilla-mobile/fenix/issues/7720 ?

**Comment by codrut-topliceanu** [comment]
@asavill Partly yes. That PR solves the first part of this issue: 

> "The switch and label should be grouped together so you hear "Enhanced Tracking Protection, Switch ON". Now those are two separate items, so you get an unlabeled switch (ie. "Switch, ON")."

I will make sure my code doesn't overwrite anything from that PR.

**Comment by asavill** [comment]
@codrut-topliceanu That's great. Thank you. Let me know If I can help :).

I also noticed this issue/PR https://github.com/mozilla-mobile/fenix/issues/8288 which seems to fix:

> The back button in the "blocked items" info sheet is unlabeled.

Apologies for the comment spam!

**Comment by codrut-topliceanu** [comment]
@asavill You're right again! I just did a rebase and the back button does have content description. I will take it into account when committing. Thank you.

**Comment by eeejay** [comment]
> @eeejay
> 
> > Each of the blocked trackers should be distinct and not be read all at once.
> 
> Do you mean it should be as a list view where the user has to navigate through the list's elements? I'm asking because there are sites were we have over a dozen or so blocked trackers.

I looked at this again. And I think it's fine as-is. The newlines allow TalkBack to pause and not read the whole thing like a giant blob. Also, reading with line granularity allows the user to read each one distinctly.

**Comment by asadotzler** [comment]
This is an accessibility P2. The feature is available and works but is difficult to use. This kind of bug prevents Fenix from being a delightful experience for users with disabilities.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest **Nightly Build 200512 from 5/12** using the following devices:
• Google Pixel 3 XL (Android 10)
• Google Pixel 3a (Android 10)
• Huawei Mate 20 Lite (Android 9)
• OnePlus A3 (Android 6.0.1)

✔️ The **switch and label are grouped together**. 
✔️ The "Blocked" heading **is marked as a heading**.  
![rsz_capturez](https://user-images.githubusercontent.com/51314259/81672832-65d6e500-9453-11ea-9548-f2e4dcdd9ca2.jpg)
✔️ Activating the blocked items goes to another screen, **the cursor will be set on the back button**.
✔️ The back button in the "blocked items" **is label as "Navigate back button"**
✔️ The "Blocked" heading in the info sheet also **has a heading role**.
✔️ Each of the blocked trackers **are read from one list, but now TalkBack to pauses between the new lines** 

► Video
![20200512-130554](https://user-images.githubusercontent.com/51314259/81673483-63c15600-9454-11ea-84db-67fd67dce129.gif)
